src/modals/SettingsModal/GeneralSettingsSection.tsx

```diff
@@ -67,14 +84,16 @@ const GeneralSettingsSection = () => {
       <KeyValueInput
         label="Password requirement"
         description="Require password confirmation before sending each transaction."
-        InputComponent={
-          <Toggle
-            toggled={passwordRequirement}
-            onToggle={() => updateSettings('general', { passwordRequirement: !passwordRequirement })}
-          />
-        }
+        InputComponent={<Toggle toggled={passwordRequirement} onToggle={onPasswordRequirementChange} />}
       />
       <HorizontalDivider narrow />
+      {requiresPasswordConfirmation && (
+        <PasswordConfirmation
+          text="Type your password to change this setting."
+          buttonText="Enter"
+          onCorrectPasswordEntered={disablePasswordRequirement}
+        />
+      )}
```

Comment:
Could you please use a modal instead (like the one when the user clicks "Show your secret phrase" in the "Wallet" tab of the "Settings" modal), so that we are more consistent with our UI patterns? This would probably help with this issue as well:

https://user-images.githubusercontent.com/1579899/166877458-6efe56e8-c089-4924-befd-c7697c4c77d4.mp4

Reply:
I thought hiding a prompt after navigating elsewhere was more desirable than always having it present...

Modal sounds good; I was worried about too many stacked modals but it's necessary if other options require password confirmations in the future!